cmd/setup-etcd-environment/run.go

```diff
@@ -129,53 +152,86 @@ func runRunCmd(cmd *cobra.Command, args []string) error {
 		out = f
 	}
 
-	if runOpts.bootstrapSRV {
-		exportEnv["DISCOVERY_SRV"] = runOpts.discoverySRV
-	} else {
-		exportEnv["NAME"] = etcdName
+	if err := writeEnvironmentFile(exportEnv, out, true); err != nil {
+		return err
 	}
 
-	// enable etcd to run using s390 and s390x. Because these are not officially supported upstream
-	// etcd requires population of environment variable ETCD_UNSUPPORTED_ARCH at runtime.
-	// https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/supported-platform.md
-	arch := runtime.GOARCH
-	if strings.HasPrefix(arch, "s390") {
-		exportEnv["UNSUPPORTED_ARCH"] = arch
+	unexportedEnv := map[string]string{
+		"IPV4_ADDRESS":      setupEnv.etcdIP,
+		"WILDCARD_DNS_NAME": fmt.Sprintf("*.%s", setupEnv.opts.discoverySRV),
 	}
-	if err := writeEnvironmentFile(exportEnv, out, true); err != nil {
-		return err
+	if setupEnv.etcdDNS != "" {
+		unexportedEnv["DNS_NAME"] = setupEnv.etcdDNS
 	}
```

Comment:
Can we please have a log entry here so we see the populated env in CI?

https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/1287/pull-ci-openshift-machine-config-operator-master-e2e-gcp-op/579/artifacts/e2e-gcp-op/pods/openshift-etcd_etcd-member-ci-op-qfg5r-m-1.c.openshift-gce-devel-ci.internal_discovery.log

Reply:
Bot exported and unexported env variables please

Reply:
I think this needs to be done in the next PR extending tests. I will note TODO.